### PR TITLE
#19 circle sort - without highlighting

### PR DESCRIPTION
**Pull request by StegSchreck**

### CODE REVIEW

File: Algorithms/CircleSort.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            return 0
+        swaps = 0
+        middle = number_of_elements // 2
+
+        for i in range(middle):
+            if self.cmp.gtI(left_pos + i, right_pos - (i + 1)):
+                self.items.swap(left_pos + i, right_pos - (i + 1))
+                swaps += 1
+
+        if (number_of_elements & 1) and self.cmp.ltI(left_pos + middle, left_pos + middle - 1):
```
Comment (Eckankar) [inline Algorithms/CircleSort.py]: I prefer `number_of_elements % 2 == 1` to `number_of_elements & 1`.

File: SortDisplay.py [hunk 2]
```diff
@@ -176,5 +177,6 @@ def update():
         print disp.citems.swaps, "swaps"
         print disp.ccmp.comparisons, "comparisons"
 
+
```
Comment (Eckankar) [inline SortDisplay.py]: Please remove this extra line. :)

File: Algorithms/CircleSort.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    def circle_sort(self, left_pos, right_pos):
+        number_of_elements = right_pos - left_pos
+        if number_of_elements < 2:
+            return 0
+        swaps = 0
+        middle = number_of_elements // 2
+
+        for i in range(middle):
+            if self.cmp.gtI(left_pos + i, right_pos - (i + 1)):
```
Comment (Eckankar) [inline Algorithms/CircleSort.py]: Whenever two elements are considered for swapping, they should be highlighted with a marker on the items.

File: Algorithms/CircleSort.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                self.items.swap(left_pos + i, right_pos - (i + 1))
+                swaps += 1
+
+        if (number_of_elements & 1) and self.cmp.ltI(left_pos + middle, left_pos + middle - 1):
+            self.items.swap(left_pos + middle - 1, left_pos + middle)
+            swaps += 1
+
+        return swaps + self.circle_sort(left_pos, left_pos + middle) + self.circle_sort(left_pos + middle, right_pos)
```
Comment (Eckankar) [inline Algorithms/CircleSort.py]: Before you do the recursion, add a marker between the two parts where you do the split.

Remember to remove the marker after the recursion is complete.

**Review by Eckankar — CHANGES_REQUESTED** [review]
I have a number of comments you need to address before I can merge it. Also, see https://github.com/Eckankar/PySortDemo/blob/master/CONTRIBUTING.md for information about how algorithms usually are structured.

**Comment by Eckankar** [comment]
#22 provided an implementation with highlighting. :)

Thanks for the effort, however.

File: SortDisplay.py (hunk 2 above)
Comment (StegSchreck) [reply SortDisplay.py:180]: You might want to have a look at PEP 8  ;)

**Comment by StegSchreck** [comment]
Good morning, I didn't had a chance to work on this the last 2 days. Sorry.

**Pull request closed without merging**